content/docs/design/conditional-concurrency/conditional-concurrency.md

```diff
@@ -0,0 +1,177 @@
+Kubegene uses the check_result function for dynamically conditional branch judging.
+
+
+switch/case example:
+
+In the following example, the result of job-a is 123, then job-b will execute, and job-c and job-d will not execute.
+
+```yaml
+Workflow:
+  Job-a:
+    tool: nginx:latest
+    commands:
+      - echo 123 # <== The execution result is reflected in std_out
+  Job-b:
+    tool: nginx:latest
+    condition: check_result(job-a, "123") # <==Judge the result of job-a and decide whether the current step is executed. Equivalent to case(123)
+    commands:
+      - echo run-job-b
+  Job-c:
+    cool: nginx:latest
+    condition: check_result(job-a, "121") # <==Just determine the result of job-a and decide whether the current step is executed. Equivalent to case(121)
+    commands:
+      - echo run-job-c
+  Job-d:
+    tool: nginx:latest
+    condition: check_result(job-a, "333") # <==Judge the result of job-a and decide whether the current step is executed. Equivalent to case (333)
+    commands:
+      - echo run-job-d
+      
+```
+The second parameter of check_result can refer to the inputs variable, such as:
+
+condition: check_result(job-a, ${my_var})
+See the check_result function for details.
+
+
+## Dependency transfer
+If the flow is A->B->C, when step B does not need to be executed because the condition is not satisfied, step C does not need to be executed 
+regardless of whether the condition is true or not due to the dependency.
+
+which is:
+
+A -> B (when condition is false ) -> C ( automatically does not need to be executed )
+
+So there is no need to write a condition condition for each job, and KubeGene will automatically determine the dependency transfer.
+
+
+# **checkresult**
+
+
+check_result is used to get the standard output of the "specified step" and determine whether it is equal to the specified string. Mainly used to control the conditional branch execution of the process.
+
+The maximum output length currently supported is 1K bytes. If the stdout of the target step is exceeded, the execution will be terminated when the execution is executed (of course, you can adjust the command after the failure and then trigger the execution).
+check_result is only available in the condition field.
+
+
+grammar
+```yaml
+Job-a:
+  tool: nginx:latest
+  commands:
+    - echo 111 # <== The output of a step
+Job-b:
+  condition: check_result(job-target, expect) # <==== Note here, determine if the output is equal
+```
+
+Example
+Suppose the function of a target step "job-a" is to judge whether the sample data is qualified or not. The standard output is:
+
+Ok
+Through the check_result function, the next step execution branch is determined.
+
+```yaml
+Job-a:
+  tool: nginx:latest
+  commands:
+    - echo ok # <== output of the step
+Job-b:
+  condition: check_result(job-a, "ok") # <==== The result here is, conditon: true
+Job-c
+  condition: check_result(job-a, "not_ok") # <==== The result here is, conditon: false
```

Comment:
one quesion： is `""` a must?

Reply:
either " " or ' ' is mandatory otherwise regexp matching will fail. I hope with the enhancement we will over come all these issues